network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/AbstractLoadingLimits.java

```diff
@@ -100,4 +99,54 @@ public double getTemporaryLimitValue(int acceptableDuration) {
         LoadingLimits.TemporaryLimit tl = getTemporaryLimit(acceptableDuration);
         return tl != null ? tl.getValue() : Double.NaN;
     }
+
+    @Override
+    public T setTemporaryLimitValue(int acceptableDuration, double temporaryLimitValue) {
+        if (temporaryLimitValue < 0 || Double.isNaN(temporaryLimitValue)) {
+            throw new ValidationException(owner, "Temporary limit value must be a positive double");
+        }
+        TemporaryLimitAttributes identifiedLimit = attributes.getTemporaryLimits() == null ? null : attributes.getTemporaryLimits().get(acceptableDuration);
+        if (identifiedLimit == null) {
+            throw new ValidationException(owner, "No temporary limit found for the given acceptable duration");
+        }
+        TreeMap<Integer, TemporaryLimitAttributes> temporaryLimitTreeMap = new TreeMap<>(attributes.getTemporaryLimits());
+        // Creation of index markers
+        Map.Entry<Integer, TemporaryLimitAttributes> biggerDurationEntry = temporaryLimitTreeMap.lowerEntry(acceptableDuration);
+        Map.Entry<Integer, TemporaryLimitAttributes> smallerDurationEntry = temporaryLimitTreeMap.higherEntry(acceptableDuration);
+
+        double oldValue = identifiedLimit.getValue();
+
+        if (isTemporaryLimitValueValid(biggerDurationEntry, smallerDurationEntry, acceptableDuration, temporaryLimitValue)) {
+            LOGGER.info("Temporary limit value changed from {} to {}", oldValue, temporaryLimitValue);
+        } else {
+            LOGGER.warn("Temporary limit value changed from {} to {}, but it is not valid", oldValue, temporaryLimitValue);
+        }
+        TemporaryLimitAttributes temporaryLimitAttributes = TemporaryLimitAttributes.builder()
+            .name(identifiedLimit.getName())
+            .value(temporaryLimitValue)
+            .acceptableDuration(acceptableDuration)
+            .operationalLimitsGroupId(identifiedLimit.getOperationalLimitsGroupId())
+            .limitType(identifiedLimit.getLimitType())
+            .build();
+        attributes.getTemporaryLimits().put(acceptableDuration, temporaryLimitAttributes);
+        return (T) this;
+    }
+
+    protected boolean isTemporaryLimitValueValid(Map.Entry<Integer, TemporaryLimitAttributes> biggerDurationEntry,
+                                                 Map.Entry<Integer, TemporaryLimitAttributes> smallerDurationEntry,
+                                                 int acceptableDuration,
+                                                 double temporaryLimitValue) {
+
+        boolean checkAgainstBigger = true;
+        boolean checkAgainstSmaller = true;
+        if (biggerDurationEntry != null) {
+            checkAgainstBigger = biggerDurationEntry.getValue().getAcceptableDuration() > acceptableDuration
+                && biggerDurationEntry.getValue().getValue() < temporaryLimitValue;
+        }
```

Comment:
we can remove redundant code and make the condition clearer like 
```  
boolean checkAgainstSmaller = biggerDurationEntry == null 
        || (biggerDurationEntry.getValue() != null 
            && biggerDurationEntry.getValue().getAcceptableDuration() > acceptableDuration
            && biggerDurationEntry.getValue().getValue() < temporaryLimitValue);
```

Reply:
checkAgainstBigger no ?